File: android/app/src/main/java/com/tisza/esemenynaptar/database/CategoryConverter.java
package com.tisza.esemenynaptar.database;

import androidx.room.*;
import com.tisza.esemenynaptar.*;

public class CategoryConverter
{
	@TypeConverter
	public static Category categoryFromString(String str)
	{
		return Category.fromStringID(str);
	}

	@TypeConverter
	public static String categoryToString(Category category)
	{
		return category == null ? null : category.getStringID();
	}

}

File: android/app/src/main/java/com/tisza/esemenynaptar/database/CategoryConverter.kt
package com.tisza.esemenynaptar.database

import androidx.room.*
import com.tisza.esemenynaptar.*

object CategoryConverter {
    @JvmStatic
	@TypeConverter
    fun categoryFromString(str: String) = Category.values().first { it.stringID == str }

    @JvmStatic
	@TypeConverter
    fun categoryToString(category: Category) = category.stringID
}
